### PR TITLE
tpcc: batch all writes in newOrder txn

### PR DESCRIPTION
**Pull request by nvb**
This change removes the iteration over each item an order in the `newOrder` transaction. It replaces
this iteration with batched operations that each work over all items in the order. This drastically reduces the number of statements that are run in the transaction and avoids O(item_count) round trips.

### CODE REVIEW

File: tpcc/new_order.go [hunk 1]
```diff
@@ -155,88 +166,165 @@ func (n newOrder) run(db *sql.DB, wID int) (interface{}, error) {
...
+			for i, item := range d.items {
+				itemIDs[i] = item.olIID
+				itemIDPls[i] = fmt.Sprintf("$%d", i+1)
 			}
-
-			selectItem, err := tx.Prepare(`SELECT i_price, i_name, i_data FROM item WHERE i_id=$1`)
+			rows, err := tx.Query(fmt.Sprintf(`
+				SELECT i_price, i_name, i_data 
+				FROM item 
+				WHERE i_id IN (%s)`, strings.Join(itemIDPls, ", ")),
```
Comment (jordanlewis) [inline tpcc/new_order.go]: I feel like you might as well just interpolate the items right into the query, like Peter was saying. But we can make that a separate patch.

File: tpcc/new_order.go [hunk 2]
```diff
@@ -155,88 +166,165 @@ func (n newOrder) run(db *sql.DB, wID int) (interface{}, error) {
...
+			)
 			if err != nil {
 				return err
 			}
+			distInfos := make([]string, d.oOlCnt)
+			sQuantityUpdateCases := make([]string, d.oOlCnt)
+			sYtdUpdateCases := make([]string, d.oOlCnt)
+			sOrderCntUpdateCases := make([]string, d.oOlCnt)
+			sRemoteCntUpdateCases := make([]string, d.oOlCnt)
+			for i := 0; rows.Next(); i++ {
```
Comment (jordanlewis) [inline tpcc/new_order.go]: I'm nervous that this is broken because the select doesn't have an order by... don't we need the results to be in the same order that we asked for from d.items?

Comment (nvb) [reply tpcc/new_order.go]: Yeah, you're completely right. I meant to fix this when I made the change to ensure that all items are unique.

Comment (nvb) [reply tpcc/new_order.go]: Done.

**Review by jordanlewis — APPROVED** [review]
LGTM but do note that adding an ORDER BY is not something that the spec asks you to do. The spec does seem flexible with regards to what actually gets executed, but I do just want to note that I'm nervous about diverging so much from what the other implementations do.

**Comment by petermattis** [comment]
We don't actually need the `ORDER BY`. We could parse each result row and store the data for the correct item. The `ORDER BY` just makes that easier.

**Comment by nvb** [comment]
Yeah the `ORDER BY` isn't strictly needed here, it just makes things a lot easier and cleaner client-side.
